apps/dashboard/src/components/workflow-editor/steps/email/views/for-view.tsx

```diff
@@ -1,36 +1,68 @@
-import { NodeViewContent, NodeViewProps, NodeViewWrapper } from '@tiptap/react';
-import { Repeat2 } from 'lucide-react';
+import { Editor, NodeViewContent, NodeViewProps, NodeViewWrapper } from '@tiptap/react';
+import { Lightbulb, Repeat2 } from 'lucide-react';
+import { VariablePill } from '@/components/variable/variable-pill';
+
+function TooltipContent() {
+  return (
+    <p className="shadow-xs absolute left-0 top-[calc(100%+8px)] flex items-center justify-center gap-2 rounded-md border border-neutral-100 bg-white px-2 py-1">
+      <Lightbulb className="size-3.5" />
+      <span className="text-xs font-medium">Use iterable variables to access the current item in the loop, e.g.</span>
+      <VariablePill
+        variableName="current."
+        hasFilters={false}
+        className="bg-bg-weak pointer-events-none text-xs font-medium"
+      />
+    </p>
+  );
+}
 
 /**
  * @see https://github.com/arikchakma/maily.to/blob/d7ea26e6b28201fc66c241200adaebc689018b03/packages/core/src/editor/nodes/for/for-view.tsx
  */
 export function ForView(props: NodeViewProps) {
   const { editor, getPos } = props;
 
+  const pos = getPos();
+  const cursorPos = editor.state.selection.from;
+
+  const forNode = editor.state.doc.nodeAt(pos);
+  const forNodeEndPos = pos + (forNode?.nodeSize ?? 0);
+
+  const isCursorInForNode = cursorPos >= pos && cursorPos <= forNodeEndPos;
+  const isOnEmptyForNodeLine = isOnEmptyLine(editor, cursorPos) && isCursorInForNode;
+
+  function isOnEmptyLine(editor: Editor, cursorPos: number) {
+    const currentLineContent = editor.state.doc
+      .textBetween(
+        Math.max(0, editor.state.doc.resolve(cursorPos).start()),
+        Math.min(editor.state.doc.content.size, editor.state.doc.resolve(cursorPos).end())
+      )
+      .trim();
+
+    return currentLineContent === '';
+  }
+
   return (
-    <NodeViewWrapper draggable="true" data-drag-handle="" data-type="repeat" className="mly-relative">
+    <NodeViewWrapper
+      draggable="true"
+      data-drag-handle=""
+      data-type="repeat"
+      className="mly-relative border-soft-100 mx-[-0.5rem] rounded-md border px-2 py-2"
+    >
       <NodeViewContent className="is-editable" />
+      {isOnEmptyForNodeLine && <TooltipContent />}
```

Comment:
seems not working, the line is not empty
![Screenshot 2025-04-14 at 12 17 02](https://github.com/user-attachments/assets/984a1a11-699c-4a59-abbe-ecb3936a52bc)